### ISSUE TITLE
"flutter version" doesn't work without a pubspec.yaml

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#51388]
This is a confusing experience to say the least:
```
$ flutter version
/home/ianh/dev/flutter/packages/flutter/lib/src: no pubspec.yaml found
$
```

**Comment by dnfield** [comment]
This was introduced by https://github.com/flutter/flutter/pull/45628, apparently to make failures make more sense.  @zanderso

**Comment by zanderso** [comment]
Near term: the `version` command needs to have its own `--no-pub` flag instead of the one that comes with `FlutterCommand`, which means something a little different.

Long term: if this were the most confusing thing about the `version` command we'd be in pretty good shape. But it isn't, so the plan is to deprecate `version` in favor of `downgrade` (which @jonahwilliams already landed here: https://github.com/flutter/flutter/pull/50506). `flutter --version` will still give the current version info.

**Comment by christopherfujino** [comment]
Closed in #51436
